userspace/plugin/plugin_api.h

```diff
@@ -939,6 +939,14 @@ typedef struct
 		//
 		ss_plugin_rc (*set_async_event_handler)(ss_plugin_t* s, ss_plugin_owner_t* owner, const ss_plugin_async_event_handler_t handler);
 	};
+
+	// Sets a new plugin configuration when provided by the framework.
+	// Required: no
+	// Arguments:
+	// - s: the plugin state, returned by init(). Can be NULL.
+	// - config: the new configuration, as an opaque string.
+	//
+	void (*set_config)(ss_plugin_t* s, const char* config);
```

Comment:
```suggestion
	ss_plugin_rc (*set_config)(ss_plugin_t* s, const char* config);
```
I think we need a way to report errors someway, take a look at how we did it in other API symbols. The plugin should have the right to reject the config and provide context on the error with `get_last_error`.

Also (not strong on this point), wouldn't it be more appropriate to pass in a `ss_plugin_set_config_input*` or something, instead than the string itself? Over time, it happened in may occasions that new features required passing more parameters than expected, so I'm always cheering for the most future-proof solution.